src/pp_tests.rs

```diff
@@ -986,17 +983,16 @@ fn parse_line() {
     // supports constant expressions)
     check_preprocessing_error("#line 1 #", PreprocessorError::UnexpectedHash);
 
-    // Test that the integer must be non-negative.
-    // TODO enabled once #line supports parsing expressions.
-    // check_preprocessing_error(
-    //     "#line -1",
-    //     PreprocessorError::LineOverflow,
-    // );
-
     // test that the integer must fit in a u32
     check_preprocessing_error("#line 4294967296u", PreprocessorError::LineOverflow);
     // test that the integer must fit in a u32
     check_preprocessed_result("#line 4294967295u", "");
+    check_preprocessing_error("#line -1", PreprocessorError::LineOverflow);
+
+    // Test some expression
+    check_preprocessed_result("#line 20 << 2 + 1", "");
```

Comment:
Maybe we could add a __LINE__ on a new line and check that it produces the correct number?